src/OrchardCore.Modules/OrchardCore.Security/Extensions/OrchardCoreBuilderExtensions.cs

```diff
@@ -0,0 +1,31 @@
+using Microsoft.Extensions.Configuration;
+using OrchardCore.Environment.Shell.Configuration;
+using OrchardCore.Security.Settings;
+
+namespace Microsoft.Extensions.DependencyInjection
+{
+    public static class OrchardCoreBuilderExtensions
+    {
+        public static OrchardCoreBuilder ConfigureSecuritySettings(this OrchardCoreBuilder builder, bool overrideAdminSettings = true)
+        {
+            builder.ConfigureServices((tenantServices, serviceProvider) =>
+            {
+                var configurationSection = serviceProvider.GetRequiredService<IShellConfiguration>().GetSection("OrchardCore_Security");
+
+                tenantServices.PostConfigure<SecuritySettings>(settings =>
+                {
+                    if (!overrideAdminSettings)
```

Comment:
At least you could put this check at the very beginning line 13 and return the builder if false, why adding a post configuration if it is for nothing. But, as did in my branch, for me this parameter is useless, if false it just means that nothing will be done, so idem if you just don't call ConfigureSecuritySettings().

About the mix of config/settings I have a couple of alternative ideas, I will comment them in the main thread.